client/src/unifyfs-sysio.c

```diff
@@ -109,42 +109,124 @@ int UNIFYFS_WRAP(mkdir)(const char* path, mode_t mode)
     }
 }
 
-int UNIFYFS_WRAP(rmdir)(const char* path)
+/* Function for shared logic between rmdir, remove, and unlink wrappers.
+ *
+ * Remove the target at upath based on the input file mode indicated by mask.
+ * If mask == 0 (e.g., remove wrapper made request), the target doesn't need
+ * any mode specific checks before attempting to remove.
+ * */
+static int unifyfs_remove_mask(const char* upath, mode_t mask)
```

Comment:
For code readability, I personally think this function should be named differently. If you just read the name, you would think it "removes a mask" from the file, which it doesn't. Perhaps just call it "unifyfs_remove"? The function comment clearly states the purpose of the mask parameter.

Reply:
Good call. Changed this.